File: projects/homelab-control-center/backend/app/schemas/platform.py
from pydantic import BaseModel


class GitState(BaseModel):
    branch: str
    commit: str


class ContainerState(BaseModel):
    running: int
    unhealthy: int


class DockerHealth(BaseModel):
    status: str


class BackupState(BaseModel):
    latest: str


class PlatformState(BaseModel):
    platform: str
    git: GitState
    containers: ContainerState
    health: DockerHealth
    backup: BackupState

File: projects/homelab-control-center/frontend/src/types/platform.ts
export interface PlatformState {
  platform: string;

  git: {
    branch: string;
    commit: string;
  };

  containers: {
    running: number;
    unhealthy: number;
  };

  health: {
    status: string;
  };

  backup: {
    latest: string;
  };
}
